src/decoder/zlib.rs

```diff
@@ -37,9 +42,21 @@ impl ZlibStream {
         self.started = false;
         self.out_buffer.clear();
         self.out_pos = 0;
+        self.max_total_output = usize::MAX;
         *self.state = Decompressor::new();
     }
 
+    pub(crate) fn set_max_total_output(&mut self, n: usize) {
+        // Add 2 and then round-up to the next multiple of 8.
+        //
+        // Increasing the original `n` helps to provide a safety margin near the end of output - in
+        // particular it helps to deal with `assert!(output.len() >= output_position + 2)` in
+        // `fdeflate::Decompressor::read`.
+        let n = n.saturating_add(2 + 7) & !7;
```

Comment:
That assert is gone in newer versions of `fdeflate` so there should be no need to round up anymore